app/Http/Controllers/PostsController.php

```diff
@@ -32,6 +32,40 @@ public function byId(Request $request)
         return self::_after($request, $post, $maxDate);
     }
     
+    public function listFeaturedPosts(Request $request)
+    {
+        $selected_number=0;
+        $selected= array();
+        $posts = self::_resolvedPosts()->orderBy('date','desc')->get();
+        foreach ($posts as $post) {
```

Comment:
You should use DB quarries instead of looping in PHP, as they are faster and get cached

Reply:
queries*